src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs

```diff
@@ -672,5 +668,95 @@ public async Task DemoteNode()
                 NoContentStatus();
             }
         }
+
+        [RavenAction("/admin/cluster/remove-entry-from-log", "POST", AuthorizationStatus.ClusterAdmin, CorsMode = CorsMode.Cluster)]
+        public async Task RemoveEntryFromLog()
+        {
+            var index = GetLongQueryString("index");
+            var first = GetBoolValueQueryString("first", false) ?? true;
+            var nodeList = new List<string>();
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                if (first)
+                {
+                    foreach (var node in ServerStore.GetClusterTopology(context).AllNodes)
+                    {
+                        if (node.Value == Server.WebUrl)
+                        {
+                            ServerStore.Engine.RemoveEntryFromRaftLog(index);
+                            nodeList.Add(node.Key);
+                            continue;
+                        }
+
+                        var cmd = new RemoveEntryFromRaftLogCommand(index);
+                        using (var requestExecutor = ClusterRequestExecutor.CreateForSingleNode(node.Value, Server.Certificate.Certificate))
+                        {
+                            await requestExecutor.ExecuteAsync(cmd, context);
+                            nodeList.AddRange(cmd.Result);
+                        }
+                    }
+                }
+                else
+                {
+                    var removed = ServerStore.Engine.RemoveEntryFromRaftLog(index);
+                    if (removed)
+                        nodeList.Add(ServerStore.NodeTag);
+                }
```

Comment:
Can simplify the code here.

```suggestion
                    var removed = ServerStore.Engine.RemoveEntryFromRaftLog(index);
                    if (removed)
                          nodeList.Add(ServerStore.NodeTag);

                if (first)
                {

                    foreach (var node in ServerStore.GetClusterTopology(context).AllNodes)
                    {
                        if (node.Value == Server.WebUrl)
                        {
                            continue;
                        }

                        var cmd = new RemoveEntryFromRaftLogCommand(index);
                        using (var requestExecutor = ClusterRequestExecutor.CreateForSingleNode(node.Value, Server.Certificate.Certificate))
                        {
                            await requestExecutor.ExecuteAsync(cmd, context);
                            nodeList.AddRange(cmd.Result);
                        }
                    }
                }
```